osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -0,0 +1,173 @@
+
+                Child = setupSkinHierarchy(() =>
+                    drawableFruit = new DrawableFruit(fruit)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(4f),
+                    }, true, false);
+            });
+
+            AddAssert("catcher custom colour", () => checkFruitHyperDashColour(drawableFruit, TestLegacySkin.CustomHyperDashColour, legacyFruit));
+        }
+
+        private Drawable setupSkinHierarchy(Func<Drawable> getChild, bool customHyperDashCatcherColour = false, bool customHyperDashFruitColour = false, bool customHyperDashAfterColour = false)
+        {
+            var testSkinProvider = new SkinProvidingContainer(new TestLegacySkin(customHyperDashCatcherColour, customHyperDashFruitColour, customHyperDashAfterColour));
+
+            var legacySkinTransformer = new SkinProvidingContainer(new CatchLegacySkinTransformer(testSkinProvider));
+
+            return testSkinProvider
+                .WithChild(legacySkinTransformer
+                    .WithChild(getChild.Invoke()));
+        }
+
+        private bool checkFruitHyperDashColour(DrawableFruit fruit, Color4 expectedColour, bool isLegacyFruit) =>
+            isLegacyFruit
+                ? fruit.ChildrenOfType<SkinnableDrawable>().First().Drawable.ChildrenOfType<Sprite>().Any(c => c.Colour == expectedColour)
+                : fruit.ChildrenOfType<SkinnableDrawable>().First().Drawable.ChildrenOfType<Circle>().Single(c => c.BorderColour == expectedColour).Any(d => d.Colour == expectedColour);
+
+        private class TestLegacySkin : ISkin
+        {
+            public static Color4 CustomHyperDashColour { get; } = Color4.Goldenrod;
+            public static Color4 CustomHyperDashFruitColour { get; } = Color4.Cyan;
+            public static Color4 CustomHyperDashAfterColour { get; } = Color4.Lime;
+
+            private readonly bool customHyperDashCatcherColour;
+            private readonly bool customHyperDashFruitColour;
+            private readonly bool customHyperDashAfterColour;
+
+            public TestLegacySkin(bool customHyperDashCatcherColour = false, bool customHyperDashFruitColour = false, bool customHyperDashAfterColour = false)
+            {
+                this.customHyperDashCatcherColour = customHyperDashCatcherColour;
+                this.customHyperDashFruitColour = customHyperDashFruitColour;
+                this.customHyperDashAfterColour = customHyperDashAfterColour;
+            }
+
+            public Drawable GetDrawableComponent(ISkinComponent component) => null;
+
+            public Texture GetTexture(string componentName)
+            {
+                if (componentName == "fruit-pear")
+                {
+                    // convince CatchLegacySkinTransformer to use the LegacyFruitPiece for pear fruit.
+                    return new Texture(Texture.WhitePixel.TextureGL)
```

Comment:
why not just return `Texture.WhitePixel` itself?

Reply:
then the fruit will be displayed in 1px

Reply:
then use a real texture or something. i thought this was just to "convince.."

Reply:
Added a default legacy skin provider instead to make things straightforward.